### DISCUSSION TITLE
Access $CONTENTSFOLD in the Unix Commands

### DISCUSSION THREAD

**Discussion by b-antwan** [The-Wineskin-Project/WineskinServer#492] [Q&A]
I would like to run some commands that use either $CONTENTSFOLD or the bundlePath in the Unix Commands section.

I need to modify the WINEPREFIX Path and set some environment variables among other things, and this is the only place I found I can do this as environment variables set in the WineryStartupScript aren't carried over to the parent (I'm guessing this is what is going on). 

I tried to do a  `set > someFile.txt` in order to check what environment variables I have at my disposal at that specific step, and the only solution I found is doing some nasty hack where I keep a substring of the WINEPREFIX or PATH. 

Is there something I missed, or some other way to get this information that doesn't require using some weird hack?

Edit: I found a better hack which involves writing the path to some file in the WineryStartupScript that I then load from the Unix Commands section. Still not ideal but better, in case this might help anyone

**Comment by emendelson** [comment]
@b-antwan - It's been years since I set this up, but I did this (and there's probably a much simpler method):

In WineryStartupScript, use something like this to set a variable and write it to a .plist file:

```
EXPORT OUTVAR=something
defaults write com.yourname.appname invar $OUTVAR
```
In Wineskin's Advanced tab, in the Unix commands field, use something like this:

```
INVAR=`defaults read com.yourname.appname invar`;
```

This is probably very stupid, but it seems to work.

EDIT: The method you described in your edit is probably more efficient than this one. Maybe the only advantage of this one is that it won't trigger permission errors when writing a file. Where do you write your file?

**Reply by b-antwan** [reply]
Thanks for the reply!
Your solution is cleaner than writing some file for sure.

I ended up doing the following:
In the WineskinStartupScript
"Calculate" and set the values of several environment variables and echo them into a file in `$HOME/tmp/someFile.env`.
(each line looks like this: `MYVAR=$Var`) 

In the Unix commands field:
Source the .env file that was created during the StartupScript, and then delete it.
`set -a; source $HOME/tmp/someFile.env; set +a; rm $HOME/tmp/someFile.env`

This way I can easily set multiple environment variables without having a long unix command field.

I'm not sure whether or not the $HOME/tmp folder exists by default, but my StartupScript creates it if it doesn't exist. Maybe a `.tmp` would be more appropriate.
